Sources/NIOSOCKS/Channel Handlers/SOCKSClientHandler.swift

```diff
@@ -31,11 +31,16 @@ public final class SOCKSClientHandler: ChannelDuplexHandler {
     private var inboundBuffer: ByteBuffer?
     
     private var bufferedWrites: MarkedCircularBuffer<(NIOAny, EventLoopPromise<Void>?)> = .init(initialCapacity: 8)
+    private var establishHandler: (Channel) -> EventLoopFuture<Void>
     
     /// Creates a new `SOCKSClientHandler` that connects to a server
     /// and instructs the server to connect to `targetAddress`.
     /// - parameter targetAddress: The desired end point - note that only IPv4, IPv6, and FQDNs are supported.
-    public init(targetAddress: SOCKSAddress) {
+    public convenience init(targetAddress: SOCKSAddress) {
+        self.init(targetAddress: targetAddress, establishHandler: { $0.eventLoop.makeSucceededFuture(()) })
+    }
+    
+    public init(targetAddress: SOCKSAddress, establishHandler: @escaping ((Channel) -> EventLoopFuture<Void>)) {
```

Comment:
Per David's note, it would be good to establish that this actually does delay removal.